src/main.rs

```diff
@@ -483,6 +498,36 @@ fn elf_to_tbf<W: Write>(
             && section.shdr.shtype == elf::types::SHT_PROGBITS
             && section.shdr.size > 0
         {
+            // This is a section we are going to add to the binary.
+
+            if last_section_address_end.is_some() {
+                // We have a previous section. Now, check if there is any
+                // padding between the sections in the .elf.
+                let end = last_section_address_end.unwrap();
+                let start = section.shdr.addr as usize;
+
+                // Because we have flash and ram memory regions, we have
+                // multiple address spaces. This check lets us assume we are in
+                // a new address segment. We need the start of the next section
+                // to be after the previous one, and the gap to not be _too_
+                // large.
+                if start > end && (start - end) < 100 {
+                    // If this is the next section in the same segment, then
+                    // check if there is any padding required.
+                    let padding = start - end;
+                    if padding > 0 {
+                        if verbose {
+                            println!("  Adding {} bytes of padding between sections", padding,);
+                        }
+
+                        // Increment our index pointer and add the padding bytes.
+                        binary_index += padding;
+                        let zero_buf = [0_u8; 512];
```

Comment:
Seems like this could be 100 instead of 512, but it shouldn't matter in practice